### PR TITLE
CI: Add Ruby 3.1, 3.0, but only for latest RSpec

### PR DESCRIPTION
**Pull request by olleolleolle**
This PR extends the build matrix with the two latest Ruby versions.

This also formats a YAML list, to make the GitHub editor not complain about formatting.

Arguable 2.7 could be in there, becoming the oldest build target.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -13,14 +13,16 @@ jobs:
...
-        ruby-version: [2.5]
-
-        rspec-core-version: [
-          3.8.2,
-          3.9.3,
-          3.10.1,
-          3.11.0,
-        ]
+        ruby-version:
+          - 3.1
```
Comment (jfragoulis) [inline .github/workflows/build.yml:17]: Why did you remove 2.5 ?

Comment (olleolleolle) [reply .github/workflows/build.yml:17]: I turned the list into the format which is best-loved by the YAML dialect in GitHub Actions. 2.5 is at the end of that list.

Comment (jfragoulis) [reply .github/workflows/build.yml:17]: Damn I am so blind. Sorry.

**Review by jfragoulis — APPROVED** [review]

File: .github/workflows/build.yml (hunk 1 above)
Comment (olleolleolle) [reply .github/workflows/build.yml:17]: How do _you_ feel about adding 2.6, perhaps 2.7?

**Comment by agis** [comment]
Thanks for re-working this to work with GitHub Actions! This looks good to me, apart from a [comment](https://github.com/skroutz/rspecq/pull/72#issuecomment-1061909913) left on the original PR (quoting here for convenience):

> One thing I'd like to do, is to only test 3.0 and 3.1 with the latest rspec core (I think it's 0.11 as of now), not everything that's in the matrix.

The reason for this is just to avoid a big number of builds that add little value. Testing only with latest rspec-core should be fine. Let's keep 2.5 tested with all rspec-core versions though.

> Arguable 2.7 could be in there, becoming the oldest build target.

Yes, let's add 2.7 as well, since it's in the "stable" set.

File: .github/workflows/build.yml (hunk 1 above)
Comment (jfragoulis) [reply .github/workflows/build.yml:17]: We don't really care about these version, as we don't use them. That being said, we should add version as needed, if needed.

**Comment by agis** [comment]
Can you rebase on top of `master`? The 3.1 failure should be fixed by https://github.com/skroutz/rspecq/commit/10071ee93a368c896ad6b94eef952d73ebc9d860 that was pushed a few minutes ago.

**Comment by agis** [comment]
Thank you!
